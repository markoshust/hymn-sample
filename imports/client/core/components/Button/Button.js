import React, { PropTypes } from 'react';

const Button = ({
  label,
  onClick,
}) => (
  <button onClick={onClick}>{label}</button>
);

Button.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  label: 'Submit',
};

export default Button;
