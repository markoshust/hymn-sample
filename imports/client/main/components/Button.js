import React, { PropTypes } from 'react';

const Button = ({
  label,
  onClick,
}) => (
  <button onClick={onClick}>{label || 'Submit'}</button>
);

Button.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Button;
