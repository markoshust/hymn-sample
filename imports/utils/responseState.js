const responseState = (response, ...state) => (
  response[state] ? response[state]() : state
);

export default responseState;
