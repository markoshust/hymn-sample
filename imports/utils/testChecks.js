import renderer from 'react-test-renderer';

const componentCheck = (component) => {
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

const renderCheck = (children) => {
  const component = renderer.create(children);

  return componentCheck(component);
};

export {
  componentCheck,
  renderCheck,
};
