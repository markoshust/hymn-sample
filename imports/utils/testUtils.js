import renderer from 'react-test-renderer';

const snap = (children) => {
  const component = renderer.create(children);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  return {
    component,
    tree,
  };
};

export {
  snap, // eslint-disable-line import/prefer-default-export
};
