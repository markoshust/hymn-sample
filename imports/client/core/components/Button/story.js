import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import Button from './Button';

const stories = storiesOf('core/Button', module);

stories.add('With Label', () => (
  <Button
    label={text('Label', 'My Button')}
    onClick={action('click')}
  />
));

stories.add('Without Label', () => (
  <Button
    onClick={action('click')}
  />
));
