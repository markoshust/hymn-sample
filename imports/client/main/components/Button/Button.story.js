import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { text } from '@kadira/storybook-addon-knobs';
import Button from './Button';

const stories = storiesOf('Button', module);

stories.add('Button', () => (
  <Button
    label={text('Label', 'My Button')}
    onClick={action('click')}
  />
));

stories.add('Without Label', () => (
  <Button
    onClick={action('click')}
  />
));
