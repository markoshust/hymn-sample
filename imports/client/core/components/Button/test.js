import React from 'react';
import { renderCheck } from '/imports/utils/testChecks';
import Button from './Button';

const onClick = jest.fn();

describe('<Button />', () => {
  it('renders correctly', () => {
    renderCheck(
      <Button
        label="My Label"
        onClick={onClick}
      />
    );
  });
});
