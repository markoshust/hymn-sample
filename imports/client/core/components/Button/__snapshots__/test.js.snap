exports[`<Button /> renders correctly 1`] = `
<button
  onClick={[Function anonymous]}>
  My Label
</button>
`;
